### ISSUE TITLE
RandZoom change the input type

### ISSUE THREAD

**Issue by drbeh** [Project-MONAI/MONAI#2920]
**Describe the bug**
When using `RandZoom` with the input type of `torch.Tensor` it will output `np.array`!

**To Reproduce**

```python
import torch
from monai.transforms import RandZoom
a = torch.randn(10, 3, 20, 20)
type(RandZoom(prob=1.0, min_zoom=0.9, max_zoom=1.1)(a))
```

**Expected behavior**
expected correct output:
```
<class 'torch.Tensor'>
```
but this will output:
```
<class 'numpy.ndarray'>
```

**Environment**
```
================================
Printing MONAI config...
================================
MONAI version: 0.6.0+117.g97fd2b92
Numpy version: 1.21.1
Pytorch version: 1.10.0a0+ecc3718
MONAI flags: HAS_EXT = False, USE_COMPILED = False
MONAI rev id: 97fd2b9201ea5e21692126539b0bddb99f1f7b23

Optional dependencies:
Pytorch Ignite version: 0.4.5
Nibabel version: 3.2.1
scikit-image version: 0.15.0
Pillow version: 8.2.0
Tensorboard version: 2.5.0
gdown version: 3.13.0
TorchVision version: 0.11.0a0
tqdm version: 4.53.0
lmdb version: 1.2.1
psutil version: 5.8.0
pandas version: 1.1.4
einops version: 0.3.0

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies


================================
Printing system config...
================================
System: Linux
Linux version: Ubuntu 20.04.2 LTS
Platform: Linux-5.8.0-59-generic-x86_64-with-glibc2.10
Processor: x86_64
Machine: x86_64
Python version: 3.8.10
Process name: python
Command: ['python', '-c', 'import monai; monai.config.print_debug_info()']
Open files: []
Num physical CPUs: 8
Num logical CPUs: 16
Num usable CPUs: 16
CPU usage (%): [1.4, 0.7, 96.5, 1.4, 3.5, 0.7, 2.1, 3.4, 0.7, 2.0, 0.7, 0.7, 0.0, 0.0, 0.0, 0.0]
CPU freq. (MHz): 1206
Load avg. in last 1, 5, 15 mins (%): [2.8, 3.6, 11.3]
Disk usage (%): 93.5
Avg. sensor temp. (Celsius): UNKNOWN for given OS
Total physical memory (GB): 31.1
Available memory (GB): 25.3
Used memory (GB): 5.3

================================
Printing GPU config...
================================
Num GPUs: 2
Has CUDA: True
CUDA version: 11.4
cuDNN enabled: True
cuDNN version: 8202
Current device: 0
Library compiled for CUDA architectures: ['sm_52', 'sm_60', 'sm_61', 'sm_70', 'sm_75', 'sm_80', 'sm_86', 'compute_86']
GPU 0 Name: NVIDIA TITAN RTX
GPU 0 Is integrated: False
GPU 0 Is multi GPU board: False
GPU 0 Multi processor count: 72
GPU 0 Total memory (GB): 23.7
GPU 0 CUDA capability (maj.min): 7.5
GPU 1 Name: NVIDIA TITAN RTX
GPU 1 Is integrated: False
GPU 1 Is multi GPU board: False
GPU 1 Multi processor count: 72
GPU 1 Total memory (GB): 23.7
GPU 1 CUDA capability (maj.min): 7.5
```

**Comment by drbeh** [comment]
@rijobro @wyli @Nic-Ma

**Comment by wyli** [comment]
had a quick look, if the padding_mode is `constant`, the outcome will be a tensor  only if the padding mode is `constant` https://github.com/Project-MONAI/MONAI/blob/281c17e549b3b1d76f585f55e1f83186ca1c8bcb/monai/transforms/croppad/array.py#L113 perhaps we can make the condition less strict here

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your analysis, let me make a quick fix ASAP.

**Comment by rijobro** [comment]
The default padding for `Zoom` is `Edge`, but torch.pad doesn't support that type of padding. See modes in https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html and https://numpy.org/doc/stable/reference/generated/numpy.pad.html.

**Comment by Nic-Ma** [comment]
> The default padding for `Zoom` is `Edge`, but torch.pad doesn't support that type of padding. See modes in https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html and https://numpy.org/doc/stable/reference/generated/numpy.pad.html.

Hi @rijobro ,

I think it's OK to use the `edge` as default value, if passing Tensor to the transform, user must set suitable mode, otherwise the transform can raise mode error.
Just like what my PR did: list all the supported modes for numpy and all the supported modes for Tensor, if not matching, raise an error.
What do you think?

Thanks.

**Comment by rijobro** [comment]
You're right that we want to minimise the amount of checks that we need to do. But this means that the `Zoom` will no longer work with torch input using the default padding. Seems like a slight backwards step to me. As mentioned in [your comment](https://github.com/Project-MONAI/MONAI/pull/2921#issuecomment-916825020), maybe our defaults should be `constant`. Generally speaking, we can try to use defaults that work for both torch and numpy.
